dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java

```diff
@@ -947,4 +949,33 @@ private String getPermissionFilterQuery(DataverseRequest dataverseRequest, SolrQ
 
     }
 
+    public String getLocaleFacetName(String name, List<DatasetFieldType> datasetFields ) {
+        final String key = StringUtils.stripAccents(name.toLowerCase().replace(" ", "_"));
```

Comment:
Maybe extract to private method + Javadoc explaining why we need to replace whitespace with _.